test/runtests.jl

```diff
@@ -50,3 +50,25 @@ end
 
     @test sum(abs.(s .- estimated)) < sum(abs.(s .- blurred_s))
 end
+
+##### Fermat Number Transform deconvolution
+
+@testset "Fermat Number Transform deconvolution tests" begin
+    using SpecialMatrices
+
+    h = [3, 2, 0, 0]
+    y = [3, 5, 3, 0]
+    x = fermat(y, h, 4, 17)
+    @test Circulant(h) * x == y
```

Comment:
Circular convolution again, it uses the same fft approach as above
https://github.com/JuliaMatrices/SpecialMatrices.jl/blob/master/src/toeplitz.jl#L54

I have feeling that we need some package with better api to perform convolution. I don't like the fact that floating point arithmetic is used here.

Reply:
It looks like you have a plan :slightly_smiling_face: